schema-to-pojo-gwt/src/test/java/org/sagebionetworks/gwt/client/schema/adapter/GwtTestSuite.java

```diff
@@ -740,4 +740,138 @@ public void testDateUtilsUTC(){
 		Date clone = DateUtils.convertStringToDate(FORMAT.UTC_MILLISEC, dateString);
 		assertEquals(now, clone);
 	}
+	
+	@Test
+	public void testArrayPutObjectWithError() throws JSONObjectAdapterException {
+		try {
+			// call under test
+			adapterArray.putObject(0, new HashSet());
+		}catch(JSONObjectAdapterException e) {
+			assertEquals("Unsupported value of type: 'java.util.HashSet' for index: '0'", e.getMessage());
+		}
+	}
+	
+	@Test
+	public void testArrayPutObjectWithString() throws JSONObjectAdapterException {
+		Object value = "someString";
+		// call under test
+		adapterArray.putObject(0, value);
+		Object result = adapterArray.getObject(0);
+		assertEquals(value, result);
+	}
+	
+	@Test
+	public void testArrayPutObjectWithNull() throws JSONObjectAdapterException {
+		Object value = null;
+		// call under test
+		adapterArray.putObject(0, value);
+		Object result = adapterArray.getObject(0);
+		assertEquals(value, result);
+	}
+	
+	@Test
+	public void testArrayPutObjectWithInteger() throws JSONObjectAdapterException {
+		Object value = new Integer(1);
+		// call under test
+		adapterArray.putObject(0, value);
+		Object result = adapterArray.getObject(0);
+		assertEquals(new Double(1.0), result);

```

Comment:
Interesting that Integers and Longs are converted to Doubles on the round-trip

Reply:
Yeah, GWT only has number().  Note: That is how it has always worked.